src/Compilers/CSharp/Portable/CSharpParseOptions.cs

```diff
@@ -67,7 +67,7 @@ public override IEnumerable<string> PreprocessorSymbolNames
             this.SpecifiedLanguageVersion = languageVersion;
             this.LanguageVersion = languageVersion.MapSpecifiedToEffectiveVersion();
             this.PreprocessorSymbols = preprocessorSymbols.ToImmutableArrayOrEmpty();
-            _features = features?.ToImmutableDictionary() ?? ImmutableDictionary<string, string>.Empty;
+            _features = features?.ToImmutableDictionary(StringComparer.OrdinalIgnoreCase) ?? ImmutableDictionary<string, string>.Empty;
```

Comment:
The default is ordinal, which doesn't work. Our suspicion is the root cause lies in a culture-specific manipulation of case somewhere prior to the use of an ordinal comparison in this method, but to date no one has conclusively identified any such location.

For me, the potentially redeeming factor with respect to this change is the fact that this particular argument is used sparingly, and it's highly unlikely a change would ever be introduced as an internal feature which has the same name as a prior feature aside from casing changes.